server/parsec/components/postgresql/utils.py

```diff
@@ -95,36 +131,46 @@ def _q(
         suffix: str | None = None,
         **kwargs: Any,
     ) -> str:
-        if table_alias:
-            from_table = f"{table} as {table_alias}"
-            select_table = table_alias
-        else:
-            from_table = table
-            select_table = table
+        # Use table alias if specified
+        from_table = f"{table} AS {table_alias}" if table_alias else table
+        select_table = table_alias if table_alias else table
+
+        # Use either internal ID (_id) or public ID for the WHERE condition
         if _id is not None:
             condition = f"{select_table}._id = {_id}"
         else:
-            public_id = kwargs.pop(public_id_field, None)
+            public_id = kwargs.pop(public_id_column, None)
             assert public_id is not None
+
+            # When using public ID, the organization should also be specified.
+            # If the organization's internal ID (`organization`) is not specified,
+            # the organization's public ID (`organization_id`) is used to retrieve it.
             if organization is None:
                 assert organization_id is not None
                 organization = q_organization_internal_id(organization_id)
-            else:
-                assert organization is not None
-            condition = f"{select_table}.organization = {organization} AND {select_table}.{ public_id_field } = { public_id }"
+
+            condition = " AND ".join(
+                (
+                    f"{select_table}.{public_id_column} = {public_id}",
+                    f"{select_table}.organization = {organization}",
+                )
+            )
+
         assert not kwargs
         suffix = suffix or ""
         return f"(SELECT {select} FROM {from_table} WHERE {condition} {suffix})"
 
     def _q_internal_id(**kwargs: Any) -> str:
         return _q(select="_id", **kwargs)
 
-    _q.__name__ = f"q_{table}"
-    _q_internal_id.__name__ = f"q_{table}_internal_id"
+    # TODO: What's the purpose of setting __name__ here? Setting __doc__ seem more useful
+    # _q.__name__ = f"q_{table}"
+    # _q_internal_id.__name__ = f"q_{table}_internal_id"
```

Comment:
I think it's useful for the function representation, but you have to use `__qualname__`
```python
In [12]: def f(): pass

In [13]: f.__qualname__= "myfunc"

In [14]: f
Out[14]: <function __main__.myfunc()>
```